### ISSUE TITLE
E275 missing whitespace after keyword for continue and break

### ISSUE THREAD

**Issue by RhythmDeolus** [muffinmad/anakin-language-server#24]
![Screenshot 2024-01-03 183739](https://github.com/muffinmad/anakin-language-server/assets/122043902/08d7ba7b-8b13-4d0e-84f9-410b0251dc1f)
![Screenshot 2024-01-03 183843](https://github.com/muffinmad/anakin-language-server/assets/122043902/8c83b787-79d1-45c5-83ff-5bbb811fb113)

**Comment by muffinmad** [comment]
I cannot reproduce this. What is the Python and pyflakes version? Can you provide source file with the minimal code to reproduce this?

**Comment by RhythmDeolus** [comment]
```python
def worker(q):
    while True:
        image = q.get()
        if image is None:
            print("Couldn't catch Image")
            continue
```
pyflakes==2.5.0
Python 3.10.9
anakinls v1.20

**Comment by muffinmad** [comment]
I see no diagnostic warnings here with the file provided

**Comment by RhythmDeolus** [comment]
I am still getting it. Do you know a way I could disable this warning alone?

**Comment by muffinmad** [comment]
Can you post the LSP events log? I'm really interested in where these warnings are coming from.

**Comment by RhythmDeolus** [comment]
~/.local/state/nvim/lsp.log
```
[START][2024-01-16 20:55:59] LSP logging initiated
[ERROR][2024-01-16 20:55:59] .../vim/lsp/rpc.lua:31     "rpc"   "/home/rhythm/.local/bin/anakinls"      "stderr"        "INFO:pygls.server:Starting IO server\n"
[ERROR][2024-01-16 20:55:59] .../vim/lsp/rpc.lua:31     "rpc"   "/home/rhythm/.local/bin/anakinls"      "stderr"        "INFO:root:Jedi environment python: /home/rhythm/anaconda3/bin/python\nINFO:root:Jedi environment sys_path:\n"
[ERROR][2024-01-16 20:55:59] .../vim/lsp/rpc.lua:31     "rpc"   "/home/rhythm/.local/bin/anakinls"      "stderr"        "INFO:root:  /home/rhythm/anaconda3/lib/python310.zip\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/lib-dynload\nINFO:root:  /home/rhythm/.local/lib/python3.10/site-packages\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages/PyQt5_sip-12.11.0-py3.10-linux-x86_64.egg\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages/mpmath-1.2.1-py3.10.egg\nINFO:root:Jedi project path: /home/rhythm/Projects/Major_Project/face_recognition_app\n"
```

**Comment by muffinmad** [comment]
Can you also provide the logs for the open file and publish diagnostics events?

**Comment by RhythmDeolus** [comment]
how do I do that?

**Comment by muffinmad** [comment]
It depends on the editor and LSP client you are using.

Below is the LSP logs from my editor. As you can see there is no warnings reported in the `publishDiagnostics` notification.
```
[internal] Tue Jan 16 17:48:11 2024:
(:message "Running language server: /Users/mad/Library/Python/3.11/bin/anakinls")
[client-request] (id:1) Tue Jan 16 17:48:11 2024:
(:jsonrpc "2.0" :id 1 :method "initialize" :params
	  (:processId 17381 :rootPath "/Users/mad/workspace/test/" :rootUri "file:///Users/mad/workspace/test" :initializationOptions #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
																		    ())
		      :capabilities
		      (:workspace
		       (:applyEdit t :executeCommand
				   (:dynamicRegistration :json-false)
				   :workspaceEdit
				   (:documentChanges t)
				   :didChangeWatchedFiles
				   (:dynamicRegistration t)
				   :symbol
				   (:dynamicRegistration :json-false)
				   :configuration t :workspaceFolders t)
		       :textDocument
		       (:synchronization
			(:dynamicRegistration :json-false :willSave t :willSaveWaitUntil t :didSave t)
			:completion
			(:dynamicRegistration :json-false :completionItem
					      (:snippetSupport :json-false :deprecatedSupport t :resolveSupport
							       (:properties
								["documentation" "details" "additionalTextEdits"])
							       :tagSupport
							       (:valueSet
								[1]))
					      :contextSupport t)
			:hover
			(:dynamicRegistration :json-false :contentFormat
					      ["plaintext"])
			:signatureHelp
			(:dynamicRegistration :json-false :signatureInformation
					      (:parameterInformation
					       (:labelOffsetSupport t)
					       :activeParameterSupport t))
			:references
			(:dynamicRegistration :json-false)
			:definition
			(:dynamicRegistration :json-false :linkSupport t)
			:declaration
			(:dynamicRegistration :json-false :linkSupport t)
			:implementation
			(:dynamicRegistration :json-false :linkSupport t)
			:typeDefinition
			(:dynamicRegistration :json-false :linkSupport t)
			:documentSymbol
			(:dynamicRegistration :json-false :hierarchicalDocumentSymbolSupport t :symbolKind
					      (:valueSet
					       [1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26]))
			:documentHighlight
			(:dynamicRegistration :json-false)
			:codeAction
			(:dynamicRegistration :json-false :codeActionLiteralSupport
					      (:codeActionKind
					       (:valueSet
						["quickfix" "refactor" "refactor.extract" "refactor.inline" "refactor.rewrite" "source" "source.organizeImports"]))
					      :isPreferredSupport t)
			:formatting
			(:dynamicRegistration :json-false)
			:rangeFormatting
			(:dynamicRegistration :json-false)
			:rename
			(:dynamicRegistration :json-false)
			:inlayHint
			(:dynamicRegistration :json-false)
			:publishDiagnostics
			(:relatedInformation :json-false :codeDescriptionSupport :json-false :tagSupport
					     (:valueSet
					      [1 2])))
		       :window
		       (:workDoneProgress t)
		       :general
		       (:positionEncodings
			["utf-32" "utf-8" "utf-16"])
		       :experimental #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
						   ()))
		      :workspaceFolders
		      [(:uri "file:///Users/mad/workspace/test" :name "~/workspace/test/")]))
[stderr] INFO:pygls.server:Starting IO server
[stderr] INFO:pygls.server:Starting IO server
[stderr] INFO:root:Jedi environment python: /opt/homebrew/opt/python@3.11/bin/python3.11
[stderr] INFO:root:Jedi environment sys_path:
[stderr] INFO:root:Jedi environment python: /opt/homebrew/opt/python@3.11/bin/python3.11
[stderr] INFO:root:Jedi environment sys_path:
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python311.zip
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
[stderr] INFO:root:  /Users/mad/Library/Python/3.11/lib/python/site-packages
[stderr] INFO:root:  /opt/homebrew/lib/python3.11/site-packages
[stderr] INFO:root:Jedi project path: /Users/mad/workspace/test
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python311.zip
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11
[stderr] INFO:root:  /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
[stderr] INFO:root:  /Users/mad/Library/Python/3.11/lib/python/site-packages
[stderr] INFO:root:  /opt/homebrew/lib/python3.11/site-packages
[stderr] INFO:root:Jedi project path: /Users/mad/workspace/test
[server-reply] (id:1) Tue Jan 16 17:48:11 2024:
(:id 1 :jsonrpc "2.0" :result
     (:capabilities
      (:positionEncoding "utf-16" :textDocumentSync
			 (:openClose t :change 2 :willSave t :willSaveWaitUntil :json-false :save
				     (:includeText :json-false))
			 :completionProvider
			 (:triggerCharacters
			  ["."])
			 :hoverProvider t :signatureHelpProvider
			 (:triggerCharacters
			  ["(" ","])
			 :definitionProvider t :referencesProvider t :documentHighlightProvider t :documentSymbolProvider t :codeActionProvider
			 (:codeActionKinds
			  ["refactor.inline" "refactor.extract"])
			 :documentFormattingProvider t :documentRangeFormattingProvider t :renameProvider t :executeCommandProvider
			 (:commands
			  [])
			 :workspace
			 (:workspaceFolders
			  (:supported t :changeNotifications t)
			  :fileOperations nil))
      :serverInfo
      (:name "anakinls" :version "1.20")))
[client-notification] Tue Jan 16 17:48:11 2024:
(:jsonrpc "2.0" :method "initialized" :params #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
							    ()))
[client-notification] Tue Jan 16 17:48:11 2024:
(:jsonrpc "2.0" :method "textDocument/didOpen" :params
	  (:textDocument
	   (:uri "file:///Users/mad/workspace/test/test.py" :version 0 :languageId "python" :text "def worker(q):\n    while True:\n        image = q.get()\n        if image is None:\n            print(\"Couldn't catch Image\")\n            continue\n")))
[client-notification] Tue Jan 16 17:48:11 2024:
(:jsonrpc "2.0" :method "workspace/didChangeConfiguration" :params
	  (:settings #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
				   ())))
[server-notification] Tue Jan 16 17:48:11 2024:
(:params
 (:uri "file:///Users/mad/workspace/test/test.py" :diagnostics
       [])
 :method "textDocument/publishDiagnostics" :jsonrpc "2.0")
```

Please consult with the LSP client documentation you are using.

**Comment by muffinmad** [comment]
Maybe this will be helpful https://neovim.discourse.group/t/where-do-language-server-logs-go/1303

**Comment by RhythmDeolus** [comment]
the lsp log file after setting the log level to debug
```

[START][2024-01-16 21:34:56] LSP logging initiated
[INFO][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:37	"Starting RPC client"	{  args = {},  cmd = "/home/rhythm/.local/bin/anakinls",  extra = {    cwd = "/home/rhythm/Projects/Major_Project/face_recognition_app"  }}
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.send"	{  id = 1,  jsonrpc = "2.0",  method = "initialize",  params = {    capabilities = {      general = {        positionEncodings = { "utf-16" }      },      textDocument = {        callHierarchy = {          dynamicRegistration = false        },        codeAction = {          codeActionLiteralSupport = {            codeActionKind = {              valueSet = { "", "quickfix", "refactor", "refactor.extract", "refactor.inline", "refactor.rewrite", "source", "source.organizeImports" }            }          },          dataSupport = true,          dynamicRegistration = true,          isPreferredSupport = true,          resolveSupport = {            properties = { "edit" }          }        },        completion = {          completionItem = {            commitCharactersSupport = true,            deprecatedSupport = true,            documentationFormat = { "markdown", "plaintext" },            insertReplaceSupport = true,            insertTextModeSupport = {              valueSet = { 1, 2 }            },            labelDetailsSupport = true,            preselectSupport = true,            resolveSupport = {              properties = { "documentation", "detail", "additionalTextEdits", "sortText", "filterText", "insertText", "textEdit", "insertTextFormat", "insertTextMode" }            },            snippetSupport = true,            tagSupport = {              valueSet = { 1 }            }          },          completionItemKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 }          },          completionList = {            itemDefaults = { "commitCharacters", "editRange", "insertTextFormat", "insertTextMode", "data" }          },          contextSupport = true,          dynamicRegistration = false,          insertTextMode = 1        },        declaration = {          linkSupport = true        },        definition = {          dynamicRegistration = true,          linkSupport = true        },        diagnostic = {          dynamicRegistration = false        },        documentHighlight = {          dynamicRegistration = false        },        documentSymbol = {          dynamicRegistration = false,          hierarchicalDocumentSymbolSupport = true,          symbolKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 }          }        },        formatting = {          dynamicRegistration = true        },        hover = {          contentFormat = { "markdown", "plaintext" },          dynamicRegistration = true        },        implementation = {          linkSupport = true        },        inlayHint = {          dynamicRegistration = true,          resolveSupport = {            properties = { "textEdits", "tooltip", "location", "command" }          }        },        publishDiagnostics = {          dataSupport = true,          relatedInformation = true,          tagSupport = {            valueSet = { 1, 2 }          }        },        rangeFormatting = {          dynamicRegistration = true        },        references = {          dynamicRegistration = false        },        rename = {          dynamicRegistration = true,          prepareSupport = true        },        semanticTokens = {          augmentsSyntaxTokens = true,          dynamicRegistration = false,          formats = { "relative" },          multilineTokenSupport = false,          overlappingTokenSupport = true,          requests = {            full = {              delta = true            },            range = false          },          serverCancelSupport = false,          tokenModifiers = { "declaration", "definition", "readonly", "static", "deprecated", "abstract", "async", "modification", "documentation", "defaultLibrary" },          tokenTypes = { "namespace", "type", "class", "enum", "interface", "struct", "typeParameter", "parameter", "variable", "property", "enumMember", "event", "function", "method", "macro", "keyword", "modifier", "comment", "string", "number", "regexp", "operator", "decorator" }        },        signatureHelp = {          dynamicRegistration = false,          signatureInformation = {            activeParameterSupport = true,            documentationFormat = { "markdown", "plaintext" },            parameterInformation = {              labelOffsetSupport = true            }          }        },        synchronization = {          didSave = true,          dynamicRegistration = false,          willSave = true,          willSaveWaitUntil = true        },        typeDefinition = {          linkSupport = true        }      },      window = {        showDocument = {          support = true        },        showMessage = {          messageActionItem = {            additionalPropertiesSupport = false          }        },        workDoneProgress = true      },      workspace = {        applyEdit = true,        configuration = true,        didChangeConfiguration = {          dynamicRegistration = false        },        didChangeWatchedFiles = {          dynamicRegistration = true,          relativePatternSupport = true        },        inlayHint = {          refreshSupport = true        },        semanticTokens = {          refreshSupport = true        },        symbol = {          dynamicRegistration = false,          symbolKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 }          }        },        workspaceEdit = {          resourceOperations = { "rename", "create", "delete" }        },        workspaceFolders = true      }    },    clientInfo = {      name = "Neovim",      version = "0.10.0-dev"    },    initializationOptions = vim.empty_dict(),    processId = 2797,    rootPath = "/home/rhythm/Projects/Major_Project/face_recognition_app",    rootUri = "file:///home/rhythm/Projects/Major_Project/face_recognition_app",    trace = "off",    workspaceFolders = { {        name = "/home/rhythm/Projects/Major_Project/face_recognition_app",        uri = "file:///home/rhythm/Projects/Major_Project/face_recognition_app"      } }  }}
[ERROR][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:31	"rpc"	"/home/rhythm/.local/bin/anakinls"	"stderr"	"INFO:pygls.server:Starting IO server\n"
[ERROR][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:31	"rpc"	"/home/rhythm/.local/bin/anakinls"	"stderr"	"INFO:root:Jedi environment python: /home/rhythm/anaconda3/bin/python\nINFO:root:Jedi environment sys_path:\n"
[ERROR][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:31	"rpc"	"/home/rhythm/.local/bin/anakinls"	"stderr"	"INFO:root:  /home/rhythm/anaconda3/lib/python310.zip\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10\n"
[ERROR][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:31	"rpc"	"/home/rhythm/.local/bin/anakinls"	"stderr"	"INFO:root:  /home/rhythm/anaconda3/lib/python3.10/lib-dynload\nINFO:root:  /home/rhythm/.local/lib/python3.10/site-packages\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages/PyQt5_sip-12.11.0-py3.10-linux-x86_64.egg\nINFO:root:  /home/rhythm/anaconda3/lib/python3.10/site-packages/mpmath-1.2.1-py3.10.egg\nINFO:root:Jedi project path: /home/rhythm/Projects/Major_Project/face_recognition_app\n"
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.receive"	{  id = 1,  jsonrpc = "2.0",  result = {    capabilities = {      codeActionProvider = {        codeActionKinds = { "refactor.inline", "refactor.extract" }      },      completionProvider = {        triggerCharacters = { "." }      },      definitionProvider = true,      documentFormattingProvider = true,      documentHighlightProvider = true,      documentRangeFormattingProvider = true,      documentSymbolProvider = true,      executeCommandProvider = {        commands = {}      },      hoverProvider = true,      positionEncoding = "utf-16",      referencesProvider = true,      renameProvider = true,      signatureHelpProvider = {        triggerCharacters = { "(", "," }      },      textDocumentSync = {        change = 2,        openClose = true,        save = {          includeText = false        },        willSave = true,        willSaveWaitUntil = false      },      workspace = {        fileOperations = vim.empty_dict(),        workspaceFolders = {          changeNotifications = true,          supported = true        }      }    },    serverInfo = {      name = "anakinls",      version = "1.20"    }  }}
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.send"	{  jsonrpc = "2.0",  method = "initialized",  params = vim.empty_dict()}
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.send"	{  jsonrpc = "2.0",  method = "workspace/didChangeConfiguration",  params = {    settings = {      anakinls = {        pyflakes_errors = { "ImportStarNotPermitted", "UndefinedExport", "UndefinedLocal", "UndefinedName", "DuplicateArgument", "MultiValueRepeatedKeyLiteral", "MultiValueRepeatedKeyVariable", "FutureFeatureNotDefined", "LateFutureImport", "ReturnOutsideFunction", "YieldOutsideFunction", "ContinueOutsideLoop", "BreakOutsideLoop", "TwoStarredExpressions", "TooManyExpressionsInStarredAssignment", "ForwardAnnotationSyntaxError", "RaiseNotImplemented", "StringDotFormatExtraPositionalArguments", "StringDotFormatExtraNamedArguments", "StringDotFormatMissingArgument", "StringDotFormatMixingAutomatic", "StringDotFormatInvalidFormat", "PercentFormatInvalidFormat", "PercentFormatMixedPositionalAndNamed", "PercentFormatUnsupportedFormat", "PercentFormatPositionalCountMismatch", "PercentFormatExtraNamedArguments", "PercentFormatMissingArgument", "PercentFormatExpectedMapping", "PercentFormatExpectedSequence", "PercentFormatStarRequiresSequence" }      }    }  }}
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.send"	{  jsonrpc = "2.0",  method = "workspace/didChangeConfiguration",  params = {    settings = {      anakinls = {        pyflakes_errors = { "ImportStarNotPermitted", "UndefinedExport", "UndefinedLocal", "UndefinedName", "DuplicateArgument", "MultiValueRepeatedKeyLiteral", "MultiValueRepeatedKeyVariable", "FutureFeatureNotDefined", "LateFutureImport", "ReturnOutsideFunction", "YieldOutsideFunction", "ContinueOutsideLoop", "BreakOutsideLoop", "TwoStarredExpressions", "TooManyExpressionsInStarredAssignment", "ForwardAnnotationSyntaxError", "RaiseNotImplemented", "StringDotFormatExtraPositionalArguments", "StringDotFormatExtraNamedArguments", "StringDotFormatMissingArgument", "StringDotFormatMixingAutomatic", "StringDotFormatInvalidFormat", "PercentFormatInvalidFormat", "PercentFormatMixedPositionalAndNamed", "PercentFormatUnsupportedFormat", "PercentFormatPositionalCountMismatch", "PercentFormatExtraNamedArguments", "PercentFormatMissingArgument", "PercentFormatExpectedMapping", "PercentFormatExpectedSequence", "PercentFormatStarRequiresSequence" }      }    }  }}
[INFO][2024-01-16 21:34:56] .../lua/vim/lsp.lua:1126	"LSP[anakin_language_server]"	"server_capabilities"	{  server_capabilities = {    codeActionProvider = {      codeActionKinds = { "refactor.inline", "refactor.extract" }    },    completionProvider = {      triggerCharacters = { "." }    },    definitionProvider = true,    documentFormattingProvider = true,    documentHighlightProvider = true,    documentRangeFormattingProvider = true,    documentSymbolProvider = true,    executeCommandProvider = {      commands = {}    },    hoverProvider = true,    positionEncoding = "utf-16",    referencesProvider = true,    renameProvider = true,    signatureHelpProvider = {      triggerCharacters = { "(", "," }    },    textDocumentSync = {      change = 2,      openClose = true,      save = {        includeText = false      },      willSave = true,      willSaveWaitUntil = false    },    workspace = {      fileOperations = vim.empty_dict(),      workspaceFolders = {        changeNotifications = true,        supported = true      }    }  }}
[DEBUG][2024-01-16 21:34:56] .../vim/lsp/rpc.lua:43	"rpc.send"	{  jsonrpc = "2.0",  method = "textDocument/didOpen",  params = {    textDocument = {      languageId = "python",      text = "from Camera import Camera\r\nfrom Display import Display\r\nfrom FaceRecog import FaceRecog\r\nimport face_recognition\r\n\r\nimport multiprocessing\r\n\r\nimport tracemalloc\r\nimport time\r\nimport datetime\r\n\r\nimport requests\r\n\r\n\r\nf = FaceRecog()\r\n\r\ncache = {\r\n  'index': [],\r\n  'value': []\r\n}\r\n\r\n\r\ndef mark_attendance(url, json):\r\n    response = requests.post(url, json=json)\r\n    print(response)\r\n    if response.status_code == 200:\r\n        return response.json()\r\n    else:\r\n        print('invalid response')\r\n    return {}\r\n\r\n\r\ndef worker(q):\r\n    url = 'http://127.0.0.1:8000/api_1/mark_attendance'\r\n    while True:\r\n        image = q.get()\r\n\r\n        if image is None:\r\n            print(\"Couldn't catch Image\")\r\n            continue\r\n        else:\r\n            frame = f.resizeImg(image, 0.25)\r\n        faces = f.detectFaces(frame)\r\n        encodings = f.getEncodings(frame, faces)\r\n        for encoding in encodings:\r\n            try:\r\n                matches = face_recognition.compare_faces(cache['index'],\r\n                                                         encoding,\r\n                                                         tolerance=0.6)\r\n                idx = matches.index(True)\r\n                print('cache hit')\r\n                print('student id: ', cache['value'][idx])\r\n            except Exception:\r\n                res = mark_attendance(url,\r\n                                      {'embedding': encoding.tolist(),\r\n                                       'time': datetime.datetime.now()\r\n                                       .strftime(\"%Y-%m-%d %H:%M:%S\")})\r\n                if ('id' in res) and res['id'] is not None:\r\n                    print('Attendance was marked for student id: ', res['id'])\r\n                    cache['index'].append(res['embedding'])\r\n                    cache['value'].append(res['id'])\r\n                else:\r\n                    print(\"couldn't find any record of you\")\r\n\r\n\r\nif __name__ == \"__main__\":\r\n    tracemalloc.start()\r\n\r\n    c = Camera()\r\n\r\n    c.open()\r\n\r\n    d = Display(0, 'Main Window')\r\n\r\n    i = 0\r\n    n = 10\r\n\r\n    q = multiprocessing.Queue()\r\n\r\n    p = multiprocessing.Process(target=worker, daemon=True, args=(q, ))\r\n    p.start()\r\n\r\n    while True:\r\n        # time.sleep(0.16)\r\n        frame1 = c.get_image()\r\n        if i % n == 0:\r\n            i = 0\r\n            if q.empty():\r\n                q.put(frame1, False)\r\n        d.displayimg(frame1)\r\n        i += 1\r\n",      uri = "file:///home/rhythm/Projects/Major_Project/face_recognition_app/Driver.py",      version = 0    }  }}
[DEBUG][2024-01-16 21:34:57] .../vim/lsp/rpc.lua:43	"rpc.receive"	{  jsonrpc = "2.0",  method = "textDocument/publishDiagnostics",  params = {    diagnostics = { {        message = "'time' imported but unused",        range = {          ["end"] = {            character = 11,            line = 8          },          start = {            character = 0,            line = 8          }        },        severity = 2,        source = "pyflakes"      }, {        code = "E275",        message = "E275 missing whitespace after keyword",        range = {          ["end"] = {            character = 20,            line = 39          },          start = {            character = 20,            line = 39          }        },        severity = 2,        source = "pycodestyle"      } },    uri = "file:///home/rhythm/Projects/Major_Project/face_recognition_app/Driver.py"  }}
```

**Comment by RhythmDeolus** [comment]
this is the whole file along with the project that I'm working on
[Driver.py](https://github.com/RhythmDeolus/MajorProject/blob/rhythm/face_recognition_app/Driver.py)
```python
from Camera import Camera
from Display import Display
from FaceRecog import FaceRecog
import face_recognition

import multiprocessing

import tracemalloc
import time
import datetime

import requests


f = FaceRecog()

cache = {
  'index': [],
  'value': []
}


def mark_attendance(url, json):
    response = requests.post(url, json=json)
    print(response)
    if response.status_code == 200:
        return response.json()
    else:
        print('invalid response')
    return {}


def worker(q):
    url = 'http://127.0.0.1:8000/api_1/mark_attendance'
    while True:
        image = q.get()

        if image is None:
            print("Couldn't catch Image")
            continue
        else:
            frame = f.resizeImg(image, 0.25)
        faces = f.detectFaces(frame)
        encodings = f.getEncodings(frame, faces)
        for encoding in encodings:
            try:
                matches = face_recognition.compare_faces(cache['index'],
                                                         encoding,
                                                         tolerance=0.6)
                idx = matches.index(True)
                print('cache hit')
                print('student id: ', cache['value'][idx])
            except Exception:
                res = mark_attendance(url,
                                      {'embedding': encoding.tolist(),
                                       'time': datetime.datetime.now()
                                       .strftime("%Y-%m-%d %H:%M:%S")})
                if ('id' in res) and res['id'] is not None:
                    print('Attendance was marked for student id: ', res['id'])
                    cache['index'].append(res['embedding'])
                    cache['value'].append(res['id'])
                else:
                    print("couldn't find any record of you")


if __name__ == "__main__":
    tracemalloc.start()

    c = Camera()

    c.open()

    d = Display(0, 'Main Window')

    i = 0
    n = 10

    q = multiprocessing.Queue()

    p = multiprocessing.Process(target=worker, daemon=True, args=(q, ))
    p.start()

    while True:
        # time.sleep(0.16)
        frame1 = c.get_image()
        if i % n == 0:
            i = 0
            if q.empty():
                q.put(frame1, False)
        d.displayimg(frame1)
        i += 1
```

**Comment by muffinmad** [comment]
Which version of `pycodestyle` do you have installed?

**Comment by RhythmDeolus** [comment]
> Which version of `pycodestyle` do you have installed?

2.10.0

**Comment by muffinmad** [comment]
The problem may be in the line endings. The line endings in the `Driver.py` file is `\r\n`. I'll investigate it further.

**Comment by muffinmad** [comment]
Should be fixed in 1.21

**Comment by RhythmDeolus** [comment]
Solved it 👍
